pyproject.toml

```diff
@@ -7,10 +7,26 @@ extend-exclude = """(\
 [tool.pytest.ini_options]
 xfail_strict = true
 
-[tool.isort]
-profile = "black"
-
 [tool.coverage.run]
 parallel = true
 branch = true
 source = ["src/taskgraph/", "taskgraph", "src/taskgraph/run-task/"]
+
+[tool.ruff]
+select = [
+    "E",    # pycodestyle
+    "F",    # pyflakes
+    "I",    # isort
+    "PLC",  # pylint convention
+    "PLE",  # pylint errors
+    "UP",   # pyupgrade
+]
```

Comment:
I suggest adding `"W"` as well
https://beta.ruff.rs/docs/rules/#warning-w

Reply:
Thinking about this. We should come up with a "releng standard set of rules for ruff"

Reply:
Good call on adding "W", codebase already passes everything there.